src/core/windows/SDL_immdevice.c

```diff
@@ -20,7 +20,7 @@
 */
 #include "../../SDL_internal.h"
 
-#if defined(__WIN32__) || defined(__GDK__)
+#if (defined(__WIN32__) || defined(__GDK__)) && HAVE_MMDEVICEAPI_H
```

Comment:
You can now add SDL_immdevice.c to Makefile.w32 for completeness:
```diff
diff --git a/Makefile.w32 b/Makefile.w32
index 1ca1fa2..978afab 100644
--- a/Makefile.w32
+++ b/Makefile.w32
@@ -76 +76 @@
-SRCS+= SDL_hid.c SDL_windows.c SDL_xinput.c
+SRCS+= SDL_hid.c SDL_immdevice.c SDL_windows.c SDL_xinput.c
```

PS: missing newline at end of SDL_immdevice.[ch]